contracts/core/Registry.sol

```diff
@@ -1,47 +1,263 @@
+    function updateIdentityName(bytes32 _identityId, string memory _name)
+        external
+        isIdentityOwner(_identityId)
+        returns (address)
+    {
+        address anchor = _generateAnchor(_identityId, _name);
+
+        Identity storage identity = identitiesById[_identityId];
+        identity.name = _name;
+
+        // clear old anchor
+        anchorToIdentityId[identity.anchor] = bytes32(0);
+
+        // set new anchor
+        identity.anchor = anchor;
+        anchorToIdentityId[anchor] = _identityId;
+
+        emit IdentityNameUpdated(_identityId, _name, anchor);
+
+        // TODO: should we return identity
+        return anchor;
+    }
+
+    /// @notice update the metadata of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _metadata The new metadata of the identity
+    /// @dev Only owner can update metadata
+    function updateIdentityMetadata(bytes32 _identityId, Metadata memory _metadata)
+        external
+        isIdentityOwner(_identityId)
+    {
+        identitiesById[_identityId].metadata = _metadata;
+
+        emit IdentityMetadataUpdated(_identityId, _metadata);
+    }
+
+    /// @notice Returns if the given address is an owner or member of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _account The address to check
+    function isOwnerOrMemberOfIdentity(bytes32 _identityId, address _account) public view returns (bool) {
+        return isOwnerOfIdentity(_identityId, _account) || isMemberOfIdentity(_identityId, _account);
+    }
+
+    /// @notice Returns if the given address is an owner of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _owner The address to check
+    function isOwnerOfIdentity(bytes32 _identityId, address _owner) public view returns (bool) {
+        return identitiesById[_identityId].owner == _owner;
+    }
+
+    /// @notice Returns if the given address is an member of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _member The address to check
+    function isMemberOfIdentity(bytes32 _identityId, address _member) public view returns (bool) {
+        return hasRole(_identityId, _member);
+    }
+
+    /// @notice Updates the pending owner of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _pendingOwner New pending owner
+    function updateIdentityPendingOwner(bytes32 _identityId, address _pendingOwner)
+        external
+        isIdentityOwner(_identityId)
+    {
+        identitiesById[_identityId].pendingOwner = _pendingOwner;
+
+        emit IdentityPendingOwnerUpdated(_identityId, _pendingOwner);
+    }
+
+    /// @notice Transfers the ownership of the identity to the pending owner
+    /// @param _identityId The identityId of the identity
+    /// @dev Only pending owner can claim ownership.
+    function changeIdentityOwner(bytes32 _identityId) external {
```

Comment:
Might name is `acceptIdentityOwnership` for clarity?

Reply:
Ah that's much better